tests/static_analysis_requirements.txt

```diff
@@ -0,0 +1,7 @@
+# the additional (!) requirements for performing static analysis of the code
+# the dependencies of `quemb` itself are given in the setup.py
+ruff
+pylint
```

Comment:
One reason why `ruff` is notably faster, is because it never calls python code.

A few checks about missing imports are much easier to do when calling python code and  `ruff` (currently) cannot perform them. That's why we have:
```yaml
    - name: Static analysis with pylint
      # ruff does nearly everything that we want from pylint
      # except for detecting unresolved imports
      # TODO: if they add it to ruff as well https://github.com/astral-sh/ruff/issues/9103
      #   remove pylint.
      run: |
        pylint --disable=all --enable=E0401,R0401,E0611 .
```

Note that `pylint` is not executed in the pre-commit hook, because of performance.

Reply:
Got it, this looks good then!